lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -346,6 +355,20 @@ data ByronWalletPostData mw = ByronWalletPostData
     , passphrase :: !(ApiT (Passphrase "encryption"))
     } deriving (Eq, Generic, Show)
 
+newtype AccountPublicKey = AccountPublicKey
+    { key :: (ApiT (ShelleyKey 'AccountK XPub))
+    } deriving (Eq, Generic, Show)
+
+newtype PostData = PostData
+    { postData :: Either WalletPostData AccountPostData
```

Comment:
So the plan is to re-use the existing endpoint? Why is this preferable to having two endpoints?

When a wallet is created using just an account public key, will that not require special treatment inside the `cardano-wallet-core`, as it has no private key? Or does existing abstractions handle this already?

And if using the same endpoint, maybe call it `WalletOrAccountPostData` instead of `PostData`? Or `WalletPostData = SomeName WalletMnemonicPostData | SomeName2 WalletAccountPostData` or something 🤷‍♂  `PostData` sounds a bit too generic at least.

Reply:
@Anviking yes, we will opt for ONE endpoint here. We talked about this with @KtorZ  and the conclusion was (when thinking about two endpoints): 
```
We are building a REST API, and we have to follow patterns and idioms of REST API. In REST, everything is about resources. Here, we are creating the same resource, so it should be the same endpoint.
```
Nevertheless, `WalletOrAccountPostData` is better than `PostData` - I agree

Reply:
> does existing abstractions handle this already?

Precisely. Attaching a private key is done as a separate operation, and, operation that requires a private key explicitly fails if there's none (e.g. signing tx)

> maybe call it WalletOrAccountPostData instead of PostData

:+1: